src/StringList.java

```diff
@@ -0,0 +1,148 @@
+import java.util.Arrays;
+public class StringList {
+    private String[] string = null;
+
+    public int size() {
+        return listNotEmpty() ? string.length : 0;
+    }
+    public boolean contains(String o) {
+        if (listNotEmpty()) {
+            for (String s : string)
+                if (o.equals(s)){
+                    return true;
+                }
+        }
+        return false;
+    }
+    public void add(String o) {
+        if (listNotEmpty()) {
+            String[] tmpArray = new String[string.length + 1];
+            System.arraycopy(string, 0, tmpArray, 0, string.length);
+            tmpArray[string.length] = o;
+            string = tmpArray;
+        } else {
+            string = new String[1];
+            string[0] = o;
+        }
+    }
+    public void add(int index, String element){         //prawdopodobnie naprawione w koncu
+        String[] tmpArray;
+        if ((listNotEmpty()) && (!outOfBounds(index))) {
+            tmpArray = new String[string.length + 1];
+            int tempIndex = 0;
+            for(int i = 0; i < tmpArray.length; i++){
+                if(i == index) {
+                    tmpArray[i] = element;
+                }
+                else{
+                    tmpArray[i] = string[tempIndex];
+                    tempIndex++;
+                }
+            }
+            string = tmpArray;
+        }
+        else if(!listNotEmpty()){
+            System.out.println("The list was empty, cannot add at specified index, adding first element to the list.");
```

Comment:
Logger zamiast sout. Poza tym mysle ze skoro lista jest pusta a nagle user chce dodac element na indeksie np 7 powinien dostac wyjatek ze indeks poza zakresem. W ogole kiedy lista jest pusta  a user daje nieprawidlowy index to user nie powinien dodac zadnego elementu a tutaj robimy mu przysluge i dodajemy na indeksie 0 co nie powinno miec miejsca

Reply:
Cóż, mój pomysł na odiotoodporność. Taka inwencja twórcza. Ale zastosuje się do porady :)